src/app/core/services/user/user.service.ts

```diff
@@ -151,11 +182,16 @@ export class UserService {
       return throwError('updateUserPassword() called without currentPassword');
     }
 
-    if (!customer.isBusinessCustomer) {
-      return this.apiService.put('customers/-/credentials/password', { password, currentPassword });
-    } else {
-      return this.apiService.put('customers/-/users/-/credentials/password', { password, currentPassword });
-    }
+    return this.appFacade.customerRestResource$.pipe(
+      concatMap(restResource =>
+        customer.isBusinessCustomer
+          ? this.apiService.put<void>('customers/-/users/-/credentials/password', { password, currentPassword })
+          : this.apiService.put<void>(`${restResource}/-/credentials/password`, {
```

Comment:
The ternary can be moved into the first argument, but I don't know if this is more readable. use your best judgement 😃